src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -933,9 +933,11 @@ inlineToHtml opts inline = do
                         linkText <- inlineListToHtml opts txt
                         slideVariant <- gets stSlideVariant
                         let s' = case s of
-                                      '#':xs | slideVariant == RevealJsSlides
-                                        -> '#':'/':xs
-                                      _ -> s
+                                   '#':xs -> let prefix = if slideVariant == RevealJsSlides
```

Comment:
I'm thinking we may need to address this at the reader level, at least in part.  If the readers added a special class to automatically generated identifiers, we could look at that in the writers.  Better, the readers could simply add the id prefix.  This would require exposing the id prefix as a reader option, too, which is a bit ugly.  Maybe the cleanest solution would be to put it in CommonState in PandocMonad, then writers and readers could use it (and pandoc.hs would have to set it inside runIO.)